code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -470,4 +471,11 @@ public void doWithLockNoResult() {
             }
         });
     }
+
+    @Override
+    public void setToken(Service service) {
+        String token = ApiKeyFilter.generateKeys()[1];
+        DataAccessor.fields(service).withKey(ServiceDiscoveryConstants.FIELD_TOKEN).set(token);
+        objectManager.persist(service);
```

Comment:
Can you use objectManager.setFields() instead.  In general avoid persisting the whole object and just set the fields you need.